deploy/gcp/main.tf

```diff
@@ -0,0 +1,336 @@
+    ports    = ["22"]
+  }
+
+  source_tags = ["bastion"]
+  target_tags = ["tidb", "tikv", "pd", "monitor"]
+}
+
+resource "google_compute_instance" "bastion" {
+  project      = "${var.GCP_PROJECT}"
+  zone         = "${var.GCP_REGION}-a"
+  machine_type = "${var.bastion_instance_type}"
+  name         = "bastion"
+
+  "boot_disk" {
+    initialize_params {
+      image = "ubuntu-os-cloud/ubuntu-1804-lts"
+    }
+  }
+
+  "network_interface" {
+    subnetwork    = "${google_compute_subnetwork.public_subnet.self_link}"
+    access_config = {}
+  }
+
+  tags = ["bastion"]
+
+  metadata_startup_script = "sudo apt-get install -y mysql-client && curl -s https://packagecloud.io/install/repositories/akopytov/sysbench/script.deb.sh | bash && sudo apt-get -y install sysbench"
+}
+
+resource "null_resource" "get-credentials" {
+  provisioner "local-exec" {
+    command = "gcloud container clusters get-credentials ${google_container_cluster.cluster.name} --region ${var.GCP_REGION}"
+
+    environment {
+      KUBECONFIG = "${local.kubeconfig}"
+    }
+  }
+
+  provisioner "local-exec" {
+    when = "destroy"
+
+    command = <<EOS
+kubectl get pvc -n tidb -o jsonpath='{.items[*].spec.volumeName}'|fmt -1 | xargs -I {} kubectl patch pv {} -p '{"spec":{"persistentVolumeReclaimPolicy":"Delete"}}'
+EOS
+
+    environment {
+      KUBECONFIG = "${local.kubeconfig}"
+    }
+  }
+}
+
+resource "local_file" "tidb-cluster-values" {
+  depends_on = ["data.template_file.tidb_cluster_values"]
+  filename   = "${local.tidb_cluster_values_path}"
+  content    = "${data.template_file.tidb_cluster_values.rendered}"
+}
+
+resource "null_resource" "setup-env" {
+  depends_on = ["google_container_cluster.cluster", "null_resource.get-credentials"]
+
+  provisioner "local-exec" {
+    working_dir = "${path.module}"
+
+    command = <<EOS
+kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $$(gcloud config get-value account)
+kubectl create serviceaccount --namespace kube-system tiller
+kubectl apply -f manifests/crd.yaml
+kubectl apply -f manifests/startup-script.yaml
+kubectl apply -f manifests/local-volume-provisioner.yaml
+kubectl apply -f manifests/gke-storage.yml
+kubectl apply -f manifests/tiller-rbac.yaml
+helm init --service-account tiller --upgrade --wait
+until helm ls; do
+  echo "Wait until tiller is ready"
+done
+helm install --namespace tidb-admin --name tidb-operator ${path.module}/charts/tidb-operator
+EOS
+
+    environment {
+      KUBECONFIG = "${local.kubeconfig}"
+    }
+  }
+}
+
+resource "null_resource" "deploy-tidb-cluster" {
+  depends_on = ["null_resource.setup-env", "local_file.tidb-cluster-values", "google_container_node_pool.pd_pool", "google_container_node_pool.tikv_pool", "google_container_node_pool.tidb_pool"]
+
+  provisioner "local-exec" {
+    command = <<EOS
+helm upgrade --install tidb-cluster ${path.module}/charts/tidb-cluster --namespace=tidb -f ${local.tidb_cluster_values_path}
```

Comment:
Terraform does not run "helm upgrade" when I change the "tidb_version" in variables.tf, add the triggers below after L318 and it works:
+  triggers {
+    values = "${data.template_file.tidb_cluster_values.rendered}"
+  }
+